cmd/bacalhau/wasm_run.go

```diff
@@ -229,43 +205,95 @@ func runWasm(
 	}
 
 	// See wazero.ModuleConfig.WithEnv
-	for key, value := range ODR.Job.Spec.Wasm.EnvironmentVariables {
+	for key, value := range ODR.EnvironmentVariables {
 		for _, str := range []string{key, value} {
 			if str == "" || strings.ContainsRune(str, null) {
 				return fmt.Errorf("invalid environment variable %s=%s", key, value)
 			}
 		}
 	}
 
+	//wasmCidOrPath := args[0]
+	entryModule, err := parseWasmEntryModule(ctx, args[0], cmd)
+	if err != nil {
+		return err
+	}
+
+	engineParams := make(map[string]interface{})
+	//ODR.Job.Spec.Wasm.Parameters = args[1:]
+	engineParams[model.WasmEngineParametersKey] = args[1:]
+	engineParams[model.WasmEngineImportModulesKey] = ODR.ImportModules
+	engineParams[model.WasmEngineEntryPointKey] = ODR.Entrypoint
+	engineParams[model.WasmEngineEntryModuleKey] = entryModule
+	engineParams[model.WasmEngineEnvVarKey] = ODR.EnvironmentVariables
+	ODR.Job.Spec.EngineSpec = model.EngineSpec{
+		Type: model.WasmEngineType,
+		Spec: engineParams,
+	}
+
 	return ExecuteJob(ctx, cm, cmd, ODR.Job, ODR.RunTimeSettings, ODR.DownloadFlags)
 }
 
+func parseWasmEntryModule(ctx context.Context, moduleArg string, cmd *cobra.Command) (model.StorageSpec, error) {
+	// Try interpreting this as a CID.
+	wasmCid, err := cid.Parse(moduleArg)
+	if err == nil {
+		// It is a valid CID – proceed to create IPFS context.
+		return model.StorageSpec{
+			StorageSource: model.StorageSourceIPFS,
+			CID:           wasmCid.String(),
+		}, nil
+	}
+	// Try interpreting this as a path.
+	info, err := os.Stat(moduleArg)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return model.StorageSpec{}, errors.Wrapf(err, "%q is not a valid CID or local file", moduleArg)
+		} else {
+			return model.StorageSpec{}, err
+		}
+	}
+
+	if !info.Mode().IsRegular() {
+		return model.StorageSpec{}, fmt.Errorf("%q should point to a single file", moduleArg)
+	}
+
+	err = os.Chdir(filepath.Dir(moduleArg))
+	if err != nil {
+		return model.StorageSpec{}, err
+	}
+
+	cmd.Printf("Uploading %q to server to execute command in context, press Ctrl+C to cancel\n", moduleArg)
+	time.Sleep(1 * time.Second)
+
+	return inline.NewStorage().Upload(ctx, info.Name())
+}
+
 func newValidateWasmCmd() *cobra.Command {
-	wasmJob := defaultWasmJobSpec()
```

Comment:
Previously we constructed the "default wasm job", populated its `Entrypoint` from the CLI and then inspect the entry point in the verification code. This changes replaces the wasm job with a string as the entrypoint is all we are verifying.